xmake/rules/c++/modules/modules_support/msvc.lua

```diff
@@ -409,35 +447,23 @@ function build_modules_for_batchcmds(target, batchcmds, objectfiles, modules, op
     local interfaceflag = get_interfaceflag(target)
     local referenceflag = get_referenceflag(target)
 
-    -- we need update mapper first
-    for _, objectfile in ipairs(objectfiles) do
-        local m = modules[objectfile]
-        if m and m.provides then
-            -- assume there that provides is only one, until we encounter the case
-            local length = 0
-            local name, provide
-            for k, v in pairs(m.provides) do
-                length = length + 1
-                name = k
-                provide = v
-                if length > 1 then
-                    raise("multiple provides are not supported now!")
-                end
-            end
-
-            local bmifile = provide.bmi
-            _add_module_to_mapper(referenceflag, name .. "=" .. path.filename(bmifile))
-            target:add("objectfiles", objectfile)
-        end
-    end
-    _flush_mapflags_to_mapper()
+    -- make sure mapper is flushed
+    _flush_mapflags_to_mapper(target)
 
-    -- append module mapper flags
-    local mapflags = _get_mapflags_from_mapper()
+    -- append target module mapper flags
+    local mapflags = _get_mapflags_from_mapper(target)
     if mapflags then
```

Comment:
it always true now.

```
    return common.localcache():get(_mapper_cachekey(target)) or {}
```

I think that `or {}` should be removed in _get_mapflags_from_mapper